### ISSUE TITLE
Feature Request: Quality selection

### ISSUE THREAD

**Issue by thegamerx1** [justfoolingaround/animdl#83]
I run animdl with a macro that takes url from browser and it would be too hard to parse output and make it choose quality so maybe a option to choose highest quality automatically or something like youtube dl does would be cool
![30_11_2021_animdl_ggERuUyLQ8](https://user-images.githubusercontent.com/34986564/144140800-a6fbcb7b-46c7-44af-8cb8-44ce02582f4c.png)

**Comment by justfoolingaround** [comment]
I'm working on a quality selector similar to youtube-dl!

This will get added soon and here's how it'll look like:

- `best`, Will select the best quality!
- `worst`, Will selects the best quality!
- `best/720/worst`, Will select the best quality, if not available, select 720, if not, select worst (or the only quality available)!
- `1080/worst`, Will select 1080, if not available, will resort to the worst quality!
- `best[!cc]/worst[cc]`, Will select the best quality that does not have a soft-subtitle, if not available (both of those constraints), will select the worst quality that has the subtitles!

By default the qualities with available subtitles will be selected (That means `[cc]` is never a necessity after all)!

**Comment by justfoolingaround** [comment]
Added and might just be overkill for anyone's needs, really.

I've made changes to the previous specifications. 

While the best and the worst haven't changed, there is an intelligent stream filter added.

That is, you can now check what is within stream metadata and make arrangements. This includes regex support.

For example:

- `best[stream_url=r"(?i)mp4upload"]`

This will select filter out stream urls that return regex matches for `(?i)mp4upload`.

- `best[title="Overlord III"][subtitle]` 

This will filter out streams titled "Overlord III" with subtitles.

You can use `/` to specify multiple segments and it'll let you know through logging as to what streams were selected by which segments.
